clang-tools/CallSites.cc

```diff
@@ -1,4 +1,13 @@
 #define _XOPEN_SOURCE 700
+
+#ifdef __GNUC__
+#pragma GCC diagnostic push
```

Comment:
if we're going to do this, let's just pass -Wno-comment on the command line.

Reply:
The idea is to prevent these warnings only when they come from headers outside our control.

Please see that I've surrounded the `#include` directives with these, and nothing else.

Reply:
Alright, I can accept that.

Reply:
Just a comment: this does not get rid of the warnings just yet. The reason is that GCC 7 has a bug that causes it to ignore this `pragma`. However, our current package dependencies fetch a newer version of gcc anyway, and in a later PR I'll force these things to be compiled with that (or clang), at which stage the warnings will actually disappear.